### PR TITLE
add readme about gateway config

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Udpate Gateway README

**Related issue(s)**

### CODE REVIEW

File: components/gateway/README.md [hunk 1]
```diff
@@ -1 +1,25 @@
...
+| APP_CONNECTOR_ORIGIN                     | http://127.0.0.1:3000           | The address and port for the director service to listen on   |
+| APP_AUDITLOG_ENABLED                     | false                           | Auditlog feature                                             |
+
+If `APP_AUDITLOG_ENABLED` is set to true, the following envs are required:
+
+| ENV                                      | Description                                                    |
+| ---------------------------------------- | -------------------------------------------------------------- |
+| APP_AUDITLOG_USER                        | The user which is used to authenticate to auditlog service     |     
+| APP_AUDITLOG_PASSWORD                    | The password which is used to authenticate to auditlog service |
+| APP_AUDITLOG_URL                         | The url where auditlog service is availables                   |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_URL**                         | The URL under which the audit log service is available                   |
```

File: components/gateway/README.md [hunk 2]
```diff
@@ -1 +1,25 @@
...
+| APP_AUDITLOG_TENANT                      | Tenant for whom the auditlogs will be logged                   |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_TENANT**                      | The tenant for whom the audit logs are logged                   |
```

File: components/gateway/README.md [hunk 3]
```diff
@@ -1 +1,25 @@
...
+| APP_AUDITLOG_CONFIG_PATH                 | The path for logging configuration changes logs                |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_CONFIG_PATH**                 | The path for logging configuration changes logs                |
```

File: components/gateway/README.md [hunk 4]
```diff
@@ -1 +1,25 @@
...
+| APP_AUDITLOG_SECURITY_PATH               | The path for loggin security events                            |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_SECURITY_PATH**               | The path for logging security events                            |
```

File: components/gateway/README.md [hunk 5]
```diff
@@ -1 +1,27 @@
 # Gateway
+
+## Overview
+
+Gateway is a component that aggregates Director's and Connector's GraphQL schemas under one service and then sends proper requests to the Director or Connector accordingly.
```
Comment (crabtree) [inline components/gateway/README.md]: It is not actually true. Gateway is a component that exposes a service through which it sends requests to backing services (Director and Connector). It does not aggregate the GraphQL schemas.

File: components/gateway/README.md [hunk 6]
```diff
@@ -1 +1,27 @@
...
+
+## Configuration
+
+Gateway binary allows you to override some configuration parameters. You can specify the following basic environment variables:
+
+| Name                             | Description                                                  | Default value            | 
+| ---------------------------------| ------------------------------------------------------------ | ------------------------ | 
+| **APP_ADDRESS**                  | The address and port for the service to listen on            | `http://127.0.0.1:3001`  |
```
Comment (crabtree) [inline components/gateway/README.md]:
```suggestion
| **APP_ADDRESS**                  | The address and port for the service to listen on            | `http://127.0.0.1:3000`  | 
```

File: components/gateway/README.md [hunk 7]
```diff
@@ -1 +1,27 @@
...
+Gateway is a component that aggregates Director's and Connector's GraphQL schemas under one service and then sends proper requests to the Director or Connector accordingly.
+
+## Configuration
+
+Gateway binary allows you to override some configuration parameters. You can specify the following basic environment variables:
+
+| Name                             | Description                                                  | Default value            | 
+| ---------------------------------| ------------------------------------------------------------ | ------------------------ | 
+| **APP_ADDRESS**                  | The address and port for the service to listen on            | `http://127.0.0.1:3001`  | 
+| **APP_DIRECTOR_ORIGIN**          | The address and port for the Director service to listen on   | `http://127.0.0.1:3000`  |
```
Comment (crabtree) [inline components/gateway/README.md]:
```suggestion
| **APP_DIRECTOR_ORIGIN**          | The address and port for the Director service to listen on   | `http://127.0.0.1:30001`  | 
```

File: components/gateway/README.md [hunk 8]
```diff
@@ -1 +1,27 @@
...
+
+## Configuration
+
+Gateway binary allows you to override some configuration parameters. You can specify the following basic environment variables:
+
+| Name                             | Description                                                  | Default value            | 
+| ---------------------------------| ------------------------------------------------------------ | ------------------------ | 
+| **APP_ADDRESS**                  | The address and port for the service to listen on            | `http://127.0.0.1:3001`  | 
+| **APP_DIRECTOR_ORIGIN**          | The address and port for the Director service to listen on   | `http://127.0.0.1:3000`  | 
+| **APP_CONNECTOR_ORIGIN**         | The address and port for the Connector service to listen on  | `http://127.0.0.1:3000`  |
```
Comment (crabtree) [inline components/gateway/README.md]:
```suggestion
| **APP_CONNECTOR_ORIGIN**         | The address and port for the Connector service to listen on  | `http://127.0.0.1:3002`  | 
```

File: components/gateway/README.md [hunk 9]
```diff
@@ -1 +1,27 @@
...
+| **APP_AUDITLOG_ENABLED**         | The variable that enables the audit log feature              | `false`                  | 
+
+If you set **APP_AUDITLOG_ENABLED** to `true`, the following environment variables are required:
+
+| Name                             | Description                                                    | 
+| -------------------------------- | -------------------------------------------------------------- |
+| **APP_AUDITLOG_USER**            | The username to the audit log service                          |
+| **APP_AUDITLOG_PASSWORD**        | The password to the audit log service                          |
+| **APP_AUDITLOG_URL**             | The URL under which the audit log service is available         |
+| **APP_AUDITLOG_TENANT**          | The tenant for whom the audit logs are logged                  |
```
Comment (crabtree) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_TENANT**          | The tenant for whom the audit logs are created                  |
```

File: components/gateway/README.md [hunk 10]
```diff
@@ -1 +1,27 @@
...
+| **APP_AUDITLOG_CONFIG_PATH**     | The path for logging configuration changes logs                |
```
Comment (crabtree) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_CONFIG_PATH**     | The path for logging configuration changes events                |
```

File: components/gateway/README.md (hunk 8 above)
Comment (dbadura) [reply components/gateway/README.md]: In config we have:
```
type config struct {
   Address string `envconfig:"default=127.0.0.1:3001"`
   DirectorOrigin  string `envconfig:"default=http://127.0.0.1:3000"`
   ConnectorOrigin string `envconfig:"default=http://127.0.0.1:3000"`
   AuditlogEnabled bool   `envconfig:"default=false"`
}
```

File: components/gateway/README.md (hunk 7 above)
Comment (dbadura) [reply components/gateway/README.md]: In config we have:
```
type config struct {
   Address string `envconfig:"default=127.0.0.1:3001"`
   DirectorOrigin  string `envconfig:"default=http://127.0.0.1:3000"`
   ConnectorOrigin string `envconfig:"default=http://127.0.0.1:3000"`
   AuditlogEnabled bool   `envconfig:"default=false"`
}
```

File: components/gateway/README.md (hunk 6 above)
Comment (dbadura) [reply components/gateway/README.md]: In config we have:
```
type config struct {
   Address string `envconfig:"default=127.0.0.1:3001"`
   DirectorOrigin  string `envconfig:"default=http://127.0.0.1:3000"`
   ConnectorOrigin string `envconfig:"default=http://127.0.0.1:3000"`
   AuditlogEnabled bool   `envconfig:"default=false"`
}
```

File: components/gateway/README.md [hunk 11]
```diff
@@ -1 +1,27 @@
 # Gateway
+
+## Overview
+
+Gateway is a component that exposes a service through which it forwards the incoming requests to backing services (Director and Connector). Optionally, Gateway can be configured to send audit logs to specified logging service.
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
Gateway is a component that exposes a service through which it forwards the incoming requests to backing services, such as the Director and Connector. Optionally, Gateway can be configured to send audit logs to the specified logging service.
```

File: components/gateway/README.md [hunk 12]
```diff
@@ -1 +1,27 @@
...
+| **APP_AUDITLOG_ENABLED**         | The variable that enables the audit log feature                   | `false`                  | 
+
+If you set **APP_AUDITLOG_ENABLED** to `true`, the following environment variables are required:
+
+| Name                             | Description                                                    | 
+| -------------------------------- | -------------------------------------------------------------- |
+| **APP_AUDITLOG_USER**            | The username to the audit log service                          |
+| **APP_AUDITLOG_PASSWORD**        | The password to the audit log service                          |
+| **APP_AUDITLOG_URL**             | The URL under which the audit log service is available         |
+| **APP_AUDITLOG_TENANT**          | The tenant for which the audit logs are created                |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_TENANT**          | The tenant for which audit logs are created                |
```

File: components/gateway/README.md [hunk 13]
```diff
@@ -1 +1,27 @@
...
+| **APP_AUDITLOG_CONFIG_PATH**     | The path for logging configuration changes events              |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_CONFIG_PATH**     | The path for events related to logging configuration changes              |
```

File: components/gateway/README.md [hunk 14]
```diff
@@ -1 +1,27 @@
...
+| **APP_AUDITLOG_SECURITY_PATH**   | The path for logging security events                           |
```
Comment (klaudiagrz) [inline components/gateway/README.md]:
```suggestion
| **APP_AUDITLOG_SECURITY_PATH**   | The path for events related to logging security                           |
```

**Pull request merged**